lib/puppetserver/ca/action/import.rb

```diff
@@ -48,41 +51,71 @@ def run(input)
 
           settings_overrides = {}
           settings_overrides[:certname] = input['certname'] unless input['certname'].empty?
+          settings_overrides[:dns_alt_names] = input['subject-alt-names'] unless input['subject-alt-names'].empty?
+
           puppet = Config::Puppet.new(config_path)
           puppet.load(settings_overrides)
           return 1 if CliParsing.handle_errors(@logger, puppet.errors)
 
-          target_locations = [puppet.settings[:cacert],
-                              puppet.settings[:cakey],
-                              puppet.settings[:cacrl],
-                              puppet.settings[:serial],
-                              puppet.settings[:cert_inventory]]
-          errors = FileSystem.check_for_existing_files(target_locations)
+          # Load most secure signing digest we can for cers/crl/csr signing.
+          signer = SigningDigest.new
+          return 1 if CliParsing.handle_errors(@logger, signer.errors)
+
+          errors = import(loader, puppet.settings, signer.digest)
+          return 1 if CliParsing.handle_errors(@logger, errors)
+
+          @logger.inform "Import succeeded. Find your files in #{puppet.settings[:cadir]}"
+          return 0
+        end
+
+        def import(loader, settings, signing_digest)
+          ca = Puppetserver::Ca::LocalCertificateAuthority.new(signing_digest, settings)
+          master_key, master_cert = ca.create_master_cert(loader.key, loader.certs.first)
+
+          FileSystem.ensure_dirs([settings[:cadir],
+                                  settings[:certdir],
+                                  settings[:privatekeydir],
+                                  settings[:publickeydir],
+                                  settings[:signeddir]])
+
+          public_files = [
+            [settings[:cacert], loader.certs],
+            [settings[:cacrl], loader.crls],
+            [settings[:localcacert], loader.certs],
+            [settings[:hostcrl], loader.crls],
+            [settings[:hostpubkey], master_key.public_key],
+            [settings[:hostcert], master_cert],
+            [settings[:cert_inventory], ca.inventory_entry(master_cert)],
+            [settings[:serial], "0x0002"],
```

Comment:
We should not in our docs, or help, that the cert/crl provided to import for the leaf CA must not have already issued or revoked any certs. Otherwise we won't know what the content of the serial and inventory files should be.

Reply:
ok!  Added it to the help text above.